test/test_torch.py

```diff
@@ -7433,12 +7433,12 @@ def test_dot_data_use(self):
         # raise a nice error.
         with self.assertRaisesRegex(
                 RuntimeError,
-                # message includes both Double and Long
-                '(?=.*Double)(?=.*Long)'):
+                # message includes both Double and ComplexFloat
+                '(?=.*Double)(?=.*ComplexFloat)'):
 
             # Calls model with a LongTensor input but DoubleTensor weights
             input = torch.randn(1, 1, 1, 6, dtype=torch.double)
-            weight = torch.zeros(1, 1, 1, 3, dtype=torch.long)
+            weight = torch.zeros(1, 1, 1, 3, dtype=torch.complex64)
```

Comment:
Why change this?

Reply:
Since `torch.long` does not support grad, with this PR, this test will fail at this assignment. However, the original test fails at `out = model(input)` with a dtype mismatch error.